UnitTests/UnitTestController.cs

```diff
@@ -0,0 +1,131 @@
+﻿using AutoMapper;
+using GetandTake.Controller;
+using GetandTake.Models.DTOs.ListDTO;
+using GetandTake.Services.Abstract;
+using Moq;
+using Xunit;
+
+namespace UnitTests;
+
+public class UnitTestController
+{
+    [Fact]
+    public void ProductList_ShouldExecuteSuccessullyAndReturnProducts()
+    {
+        //arrange
+        var productService = new Mock<IProductService>();
+        var productList = GetProducts();
+        productService.Setup(product => product.GetAllAsync())
+            .ReturnsAsync(productList);
+        var productController = new ProductController(productService.Object);
+
+        //act
+        var productResult = productController.ProductList().Result;
+
+        //assert
+        Assert.NotNull(productResult);
+        Assert.Equal(productList.Count, productResult.Count);
+    }
+
+    [Fact]
+    public void Product_ShouldExecuteSuccessullyAndReturnProduct()
+    {
+        //arrange
+        var productService = new Mock<IProductService>();
+        var productList = GetProducts();
+        productService.Setup(product => product.GetByIdAsync(2))
+            .ReturnsAsync(productList[1]);
+        var productController = new ProductController(productService.Object);
+
+        //act
+        var productResult = productController.GetProductByIdAsync(2);
+
+        //assert
+        Assert.NotNull(productResult);
+        Assert.Equal(productList[1].ProductID, productResult.Result.ProductID);
+    }
+
+    [Fact]
+    public void Product_GetProductByIdWithInvalıdArguments_ThrowsArgumentException()
+    {
+        //arrange
+        var productService = new Mock<IProductService>();
+        var productList = GetProducts();
+        productService.SetupSequence(product => product.GetByIdAsync(8))
```

Comment:
No need to set sequence, you can just call `Setup()` and `ThrowsAsync(new Exception())` after that, behaviour will be the same. Also it has some unreal case: your method is returning result and throws an exception, this never happens in real life, only one scenario :)